### PR TITLE
Reorganize the C# Guide around a new C# hub page

### LINKED ISSUE #39820: Create C# hub page
C# need a hub page, instead of a landing page for more visibility with our training materials. 

In addition to the Hub page, the following updates need to be made to some docs infrastructure:

1. Create a C# L2 header for the C# guide. This should be similar to the .NET and .NET aspire headers, so that C# is still tightly integrated with the full .NET ecosystem.
2. Update the breadcrumbs for the C# guide. The home of the breadcrumb should be Learn / C#

By creating a hub page, we can use landing pages in the different areas for the C# Guide. I don't know how many we'll want in the end, but we should start with a "main" section and a "specification" section. That will minimize the number of readers that end up in the wrong location.

---
[Associated WorkItem - 226916](https://dev.azure.com/msft-skilling/Content/_workitems/edit/226916)

### LINKED ISSUE #40154: Create C# Hub page
In order to add C# as a devlang on the training site, C# needs to have a hub page rather than a landing page. As doing that, take care of the following:

- Create a C# hub page
- Create landing page for the C# language reference
- Create a landing page for the C# specification
- Create sub-toc for Language reference
- Create sub-toc for C# specification.

### PR DESCRIPTION
**Pull request by BillWagner**
Fixes #39820
Fixes #40154 

1. Create sub-TOC for specifications, along with a landing page for language specs.
1. Create sub-TOC for language reference, along with a landing page for the language reference. This required also updating or removing links to the previous index page. The links to the index was removed from pages in the language reference. In other areas, the link was updated to point at the new YML file.
1. Incorporate compiler errors messages into the language reference TOC.
1. Create new C# hub page.

Primary files changed:

- [New C# Hub page](https://review.learn.microsoft.com/en-us/dotnet/csharp/?branch=pr-en-us-39916)
- [Language reference landing page](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/?branch=pr-en-us-39916)
- [C# Language Spec landing page](https://review.learn.microsoft.com/en-us/dotnet/csharp/specification/?branch=pr-en-us-39916)

The two new landing pages move that content out of the main conceptual TOC, which makes it more stand-alone. readers are less likely to fall into it by accident.

Most of the remaining files changed by removing a link to the overview page of the language reference. That seems redundant for language reference pages, as that page is the top page in the TOC.

---

#### Internal previews

<details><summary><strong>Toggle expand/collapse</strong></summary><br/>

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/csharp/fundamentals/exceptions/exception-handling.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/fundamentals/exceptions/exception-handling.md) | [Exception Handling (C# Programming Guide)](https://review.learn.microsoft.com/en-us/dotnet/csharp/fundamentals/exceptions/exception-handling?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/bool.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/bool.md) | ["bool type - C# reference"](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/bool?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/built-in-types.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/built-in-types.md) | [Built-in types (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/built-in-types?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/char.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/char.md) | [char (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/char?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/default-values.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/default-values.md) | [docs/csharp/language-reference/builtin-types/default-values](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/default-values?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/enum.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/enum.md) | [Enumeration types (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/enum?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/floating-point-numeric-types.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/floating-point-numeric-types.md) | [Floating-point numeric types (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/floating-point-numeric-types?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/integral-numeric-types.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/integral-numeric-types.md) | [Integral numeric types  (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/integral-numeric-types?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/nullable-reference-types.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/nullable-reference-types.md) | ["Nullable reference types - C# reference"](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-reference-types?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/nullable-value-types.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/nullable-value-types.md) | [Nullable value types (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/nullable-value-types?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/numeric-conversions.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/numeric-conversions.md) | [Built-in numeric conversions (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/numeric-conversions?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/record.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/record.md) | [docs/csharp/language-reference/builtin-types/record](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/record?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/ref-struct.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/ref-struct.md) | [`ref` structure types (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/ref-struct?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/reference-types.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/reference-types.md) | [Built-in reference types (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/reference-types?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/struct.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/struct.md) | ["Structure types - C# reference"](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/struct?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/unmanaged-types.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/unmanaged-types.md) | [Unmanaged types (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/unmanaged-types?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/value-tuples.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/value-tuples.md) | [Tuple types (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-tuples?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/value-types.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/value-types.md) | [Value types (C# reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/value-types?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/builtin-types/void.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/builtin-types/void.md) | ["void - C# reference"](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/builtin-types/void?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/keywords/abstract.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/keywords/abstract.md) | [abstract (C# Reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/abstract?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/keywords/access-modifiers.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/keywords/access-modifiers.md) | [docs/csharp/language-reference/keywords/access-modifiers](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/access-modifiers?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/keywords/accessibility-domain.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/keywords/accessibility-domain.md) | [Accessibility Domain (C# Reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/accessibility-domain?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/keywords/accessibility-levels.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/keywords/accessibility-levels.md) | [Accessibility Levels (C# Reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/accessibility-levels?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/keywords/ascending.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/keywords/ascending.md) | ["ascending - C# Reference"](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/ascending?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/keywords/index.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/keywords/index.md) | [C# Keywords](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/index?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/keywords/out.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/keywords/out.md) | [out (C# Reference)](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/out?branch=pr-en-us-39916) |
| [docs/csharp/language-reference/tokens/index.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/language-reference/tokens/index.md) | [C# Special Characters](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/index?branch=pr-en-us-39916) |
| [docs/csharp/misc/sorry-we-don-t-have-specifics-on-this-csharp-error.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/misc/sorry-we-don-t-have-specifics-on-this-csharp-error.md) | [Sorry, we don't have specifics on this C# error](https://review.learn.microsoft.com/en-us/dotnet/csharp/misc/sorry-we-don-t-have-specifics-on-this-csharp-error?branch=pr-en-us-39916) |
| [docs/csharp/programming-guide/classes-and-structs/implicitly-typed-local-variables.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/programming-guide/classes-and-structs/implicitly-typed-local-variables.md) | [docs/csharp/programming-guide/classes-and-structs/implicitly-typed-local-variables](https://review.learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/implicitly-typed-local-variables?branch=pr-en-us-39916) |
| [docs/csharp/specification/overview.md](https://github.com/dotnet/docs/blob/17f0741483056ab5bd7b30570956b0a9a7932958/docs/csharp/specification/overview.md) | [C# standard specification](https://review.learn.microsoft.com/en-us/dotnet/csharp/specification/overview?branch=pr-en-us-39916) |

</details>

> **Note**
> This table shows preview links for the 30 files with the most changes. For preview links for other files in this PR, select <strong>OpenPublishing.Build Details</strong> within [checks](https://github.com/dotnet/docs/pull/39916/checks).

### CODE REVIEW

**Comment by katiesavage** [comment]
I also wonder if we should add the 'concepts' from the original C# index page into the "language concepts" section: 
image

**Comment by BillWagner** [comment]
The MS Docs build verifier error is false positive: It's because `index.md` changed to `index.yml` in the C# root folder. Adding a redirection for that fails the OPS build.

**Comment by BillWagner** [comment]
@jamesmontemagno @claudiaregio @katiesavage @KathleenDollard 

This is ready to merge. I tagged all of you for review if you want to take another look.

File: docs/csharp/index.yml [hunk 1]
```diff
@@ -1,162 +1,338 @@
-### YamlMime:Landing
+### YamlMime:Hub
 
-title: "C# documentation"
-summary: "Learn how to write any application using the C# programming language on the .NET platform."
+#root section (Required)
+title: C# language documentation
+summary: >
+  "The C# guide contains articles, tutorials, and code samples to help you get started with C# and the .NET platform. Experienced developers can learn about new
```
Comment (katiesavage) [inline docs/csharp/index.yml:6]:
```suggestion
  The C# guide contains articles, tutorials, and code samples to help you get started with C# and the .NET platform. Experienced developers can learn about new
```

File: docs/csharp/index.yml [hunk 2]
```diff
@@ -1,162 +1,338 @@
...
+  features in the What's new section. Experienced developers can learn details of language behavior from the reference and language specifications."
```
Comment (katiesavage) [inline docs/csharp/index.yml:7]:
```suggestion
  features in the What's new section. Experienced developers can learn details of language behavior from the reference and language specifications.
```

File: docs/csharp/index.yml [hunk 3]
```diff
@@ -1,162 +1,338 @@
...
+      links:
+        - url: tour-of-csharp/index.md
+          text: A tour of C#
+        - url: ./tour-of-csharp/tutorials/index.md
+          text: Beginner C# tutorials
+        - url: ./tour-of-csharp/tutorials/hello-world.yml
+          text: Try C# in your browser
+        - url: ./fundamentals/program-structure/index.md
+          text: "Inside a C# program"
+
```
Comment (katiesavage) [inline docs/csharp/index.yml:68]: Can we add the learning resources here as well?
- C# for Beginners video series: https://aka.ms/dotnet/beginnervideos/learn/csharp
- Foundational C# Certification: https://aka.ms/csharp-certification

File: docs/csharp/index.yml [hunk 4]
```diff
@@ -1,162 +1,338 @@
...
+    # Card
+    - title: Learn C# video series
+      itemType: video
+      url: https://aka.ms/dotnet/beginnervideos/learn/csharp
+    # Card
+    - title: C# Learning paths
+      itemType: training
+      url: https://learn.microsoft.com/en-us/collections/yz26f8y64n7k07
+    # Card
+    - title: C# Certification - Free code camp
```
Comment (katiesavage) [inline docs/csharp/index.yml:45]:
```suggestion
    - title: Foundational C# Certification - freeCodeCamp
```

File: docs/csharp/index.yml [hunk 5]
```diff
@@ -1,162 +1,338 @@
...
+      itemType: whats-new
+      url: ./whats-new/csharp-13.md
+    # Card
+    - title: Learn C# video series
+      itemType: video
+      url: https://aka.ms/dotnet/beginnervideos/learn/csharp
+    # Card
+    - title: C# Learning paths
+      itemType: training
+      url: https://learn.microsoft.com/en-us/collections/yz26f8y64n7k07
```
Comment (katiesavage) [inline docs/csharp/index.yml:43]:
```suggestion
      url: https://learn.microsoft.com/collections/yz26f8y64n7k07
```

File: docs/csharp/index.yml [hunk 6]
```diff
@@ -1,162 +1,342 @@
...
+              itemType: reference
+              text: C# draft standard
+            - url: ../../_csharplang/proposals/csharp-10.0/enhanced-line-directives.md
+              itemType: reference
+              text: Feature specifications
+
+# tools section (Optional; Remove if not applicable.)
+tools:
+  title: Build C# apps with the Visual Studio family
+  summary: Choose Visual Studio or VS Code to build your C# applications.
```
Comment (gewarren) [inline docs/csharp/index.yml:190]:
```suggestion
  summary: Choose Visual Studio or Visual Studio Code to build your C# applications.
```

**Review by gewarren — APPROVED** [review]
Looks great!

File: docs/csharp/index.yml [hunk 7]
```diff
@@ -1,162 +1,342 @@
...
+        # Card
+        - title: C# specification
+          summary: The official specification for the C# language
+          links:
+            - url: ./specification/index.yml
+              itemType: reference
+              text: Standard and specifications
+            - url: ./specification/overview.md
+              itemType: reference
+              text: linktext1
```
Comment (BillWagner) [inline docs/csharp/index.yml:179]:
```suggestion
              text: The specification process
```

**Comment by BillWagner** [comment]
Thanks a lot @gewarren That was awesome!

**Pull request merged**